docs/quickstart.md

```diff
@@ -0,0 +1,108 @@
+# Quickstart
+
+Simple-JSON was designed to be the simplest way possible for PureScript users to safely consume untyped, JSON structured data.
+
+## Reading and Writing Simple Types
+
+Simple-JSON can be used to read and write simple types, such as numbers, ints, strings, booleans, etc.
+
+For example, the snippet below attempts to parse an integer from the string `"1"` and print it to the console:
+```hs
+import Prelude
+import Data.Either (Either(..))
+import Effect (Effect)
+import Effect.Console
+import Simple.JSON as JSON
+
+main :: Effect Unit
+main =
+  case (JSON.readJSON "1") of
+    Right (int :: Int) -> 
+      log ("Parsed the integer " <> show int)
+    Left errs -> 
+      log ("Failed to parse the input as an integer: " <> show errs)
+```
+
+Note that, because `JSON.readJSON` returns `Either MultipleErrors a`, the `Right` pattern match must be explicitly annotated in this case, since there's no other way to infer that we _actually_ want to parse out an `Int` here.
+
+Writing simple types out as JSON strings is even easier:
+```hs
+import Prelude
+import Effect (Effect)
+import Effect.Console
+import Simple.JSON as JSON
+
+main :: Effect Unit
+main =
+  let stringifiedInt = JSON.writeJSON (1 :: Int)
+  in log ("Writing the integer: " <> stringifiedInt)
+```
+
+Reading and writing non-integer simple types is just as easy, just keep in mind that there might not always be enough information available for the compiler to infer _precisely_ what you want to read and write.
+
+When in doubt, annotate the arguments and results of functions as appropriate!
+
+## Handling Optionality
+
+Simple-JSON provides functions that can deal with the two types of optionality one might encounter when dealing with JSON data: `null` and `undefined`.
+
+To disambiguate between these two cases, Simple-JSON relies on the the `Nullable` newtype to represent fields that might be `null`, and uses the standard `Maybe` type tp represent fields that might be `undefined`.
```

Comment:
I think somehow this might be reworded as most users will probably just want `Maybe` by default

Reply:
Any thoughts? I wouldn't have thought people would prefer `undefined` by default, which is why I worded it like this in the first place.

Maybe something like
```
By default `Simple-JSON` will encode `Nothing :: Maybe a` as `undefined` when reading and writing JSON, however this behavior can be modified by wrapping `Maybe` fields in the `Nullable` newtype which is designed to interoperate with JavaScript's `null`s.
```